core/src/main/scala/kafka/log/LogCleanerManager.scala

```diff
@@ -345,16 +347,57 @@ private[log] object LogCleanerManager extends Logging {
     log.config.compact && log.config.delete
   }
 
+  /**
+    * return the baseOffset of the first dirty segment that was created at or after cleanUtilTime
+    */
+  def getMustCleanOffset(log: Log, firstDirtyOffset: Long, cleanUtilTime: Long) : Long = {
+
+    val largestCleanTimestamp = log.logSegments(-1, firstDirtyOffset).lastOption match {
```

Comment:
Nit: 
```
val largestCleanTimestamp = log.logSegments(-1, firstDirtyOffset).lastOption.map(_.largestTimestamp).getOrElse(0L) 
```
one line can do this, but I guess pattern matching is preferable in some cases :)

Reply:
This looks much simpler and more scala-like. fixed.